src/Compilers/Shared/RuntimeHostInfo.cs

```diff
@@ -58,18 +57,28 @@ internal static (string processFilePath, string commandLineArguments, string too
         /// </summary>
         internal static string GetDotNetPathOrDefault()
         {
-            if (Environment.GetEnvironmentVariable(DotNetHostPathEnvironmentName) is string pathToDotNet)
+            if (Environment.GetEnvironmentVariable(DotNetHostPathEnvironmentName) is { Length: > 0 } pathToDotNet)
             {
                 return pathToDotNet;
             }
 
+            if (Environment.GetEnvironmentVariable(DotNetExperimentalHostPathEnvironmentName) is { Length: > 0 } pathToDotNetExperimental)
+            {
+                return pathToDotNetExperimental;
+            }
+
             var (fileName, sep) = PlatformInformation.IsWindows
                 ? ("dotnet.exe", ';')
                 : ("dotnet", ':');
 
             var path = Environment.GetEnvironmentVariable("PATH") ?? "";
-            foreach (var item in path.Split(sep, StringSplitOptions.RemoveEmptyEntries))
+            foreach (var item in path.Split(sep))
             {
+                if (string.IsNullOrWhiteSpace(item))
+                {
+                    continue;
+                }
```

Comment:
Why did you split this out?

Reply:
You're right, that shouldn't be needed, thanks.